rfcs/0007-consistent-vehicle-ids.md

```diff
@@ -0,0 +1,185 @@
+- Light Rail (Green Line and Mattapan)
+
+Within and between these modes, the identifiers of the vehicle (printed on
+the outside) can overlap, causing confusion in our real-time data. By
+creating consistent prefixes for each mode, we can ensure that the
+identifiers used in our real-time data are unique across all modes.
+
+# Guide-level explanation
+[guide-level-explanation]: #guide-level-explanation
+
+<!--
+Explain the proposal as if it was already implemented and you were teaching it to a new developer
+that just joined the team. That generally means:
+
+- Introducing new named concepts.
+- Explaining the feature largely in terms of examples.
+- Explaining how programmers should *think* about the feature, and how it should impact the way they
+  work on this project. It should explain the impact as concretely as possible.
+- If applicable, provide sample error messages, deprecation warnings, or migration guidance.
+- If applicable, describe the differences between teaching this to senior developers and to junior
+  developers. -->
+
+Our vehicle information is published in our GTFS-RT VehiclePositions file. There are three fields which currently contain some vehicle information:
+
+| Field | Description | Values |
+| --- | --- | --- |
+| `id` | Internal system of identification for the vehicle. Should be unique to the vehicle | Heavy/light rail: the line the car is on plus a hex code that is unique for the consist<br>Commuter Rail: the number of the control coach<br>Bus: the letter "y" plus the number of the bus |
+| `label` | a user visible label - for example the name of a train | Heavy/light rail: the number of the lead car<br>Commuter Rail: the number of the control coach<br>Bus: the number of the bus |
+| `consist` | Information about a single car in a multi-car train | Heavy/rail: `label` contains the number of each car making up the train, in order from front to back |
+
+In particular, bus labels and commuter rail labels overlap, leading us to
+prefix the bus IDs with a `y` to distinguish them. This works as long as we
+only have one type of bus information. However, if we add additional sources
+of bus data (such as contracted shuttle buses) we need another prefix in
+order to distinguish the new vehicles.
+
+To avoid this, this RFC introduces consistent prefixes to give each group of vehicles a unique namespace. The pattern is
+`<vendor>-<mode/line>-<unique identifier>`.
+
+| Mode / Line | Prefix | Old => New |
+| --- | --- | --- |
+| Red Line | `mbta-red` | `R-546C3C33` => `mbta-red-546C3C33` |
+| Orange Line | `mbta-orange` | `O-546C3BA5` => `mbta-orange-546C3BA5` |
+| Green Line / Mattapan High-Speed Rail | `mbta-green` | `G-10100` => `mbta-green-10100` |
+| Blue Line | `mbta-blue` | `B-546C3C08` => `mbta-blue-546C3C08` |
+| Commuter Rail | `keolis-rail` | `1809` => `keolis-rail-1809`
+| MBTA Bus | `mbta-bus` | `y1454` => `mbta-bus-1454` |
+| Yankee Shuttle Bus | `yankee-bus` | `ynk1234` => `yankee-bus-1234` |
```

Comment:
**suggestion (non-blocking):** not sure if it makes sense to add in this section or elsewhere, but it might be worth considering how this would work for ferry should we ever get real-time ferry location data.